cryptographic_estimators/base_estimator.py

```
@@ -328,37 +271,12 @@ def table(self, show_quantum_complexity=0, show_tilde_o_time=0, show_all_paramet
         - ``truncate`` -- truncate rather than round the output (default: false)
 
         """
-
         self.include_tildeo = show_tilde_o_time
         self.include_quantum = show_quantum_complexity
-        est = self.estimate()
-
-        if est is None:
-            print("No algorithms associated with this estimator.")
-        key = list(est.keys())[0]
-        tables = []
-        tbl = self._create_initial_table_containing_algorithm_column()
-        tables.append(tbl)
-
-        for j in est[key].keys().__reversed__():
-            if j == BASE_QUANTUMO and not show_quantum_complexity:
-                continue
-            if j == BASE_TILDEO_ESTIMATE and not show_tilde_o_time:
-                continue
-            if j == BASE_ADDITIONALO:
-                continue
-
-            tbl = self._create_subtable_containing_all_columns(
-                j, show_all_parameters)
-            self._add_rows(tbl, truncate, precision)
-            tables.append(tbl)
-
-            if j == BASE_QUANTUMO:
-                tbl._min_width = {BASE_TIME: len(BASE_QUANTUMO)}
-
-        tbl_join = concat_all_tables(tables)
-
-        print(tbl_join)
+        renderer = EstimationRenderer(
+            show_quantum_complexity, show_tilde_o_time, show_all_parameters, precision, truncate
+        )
+        renderer.as_table(self.estimate())
```

it should estimate first and if the return of estimate is an empty dictionary it should throw an error "No algorithm associated with this estimator or applicable to this problem instance"

---

if not empty it can be passed to the renderer

---

Ok thanks, I will fix this